lessons/specifics/iex_helpers.md

```diff
@@ -0,0 +1,228 @@
+---
+layout: page
+title: IEx Helpers
+category: specifics
+order: 6
+lang: en
+---
+
+{% include toc.html %}
+
+## Overview
+As you begin to work in Elixir, IEx is your best friend.
+It is a REPL, but it has many advanced features that can make life easier when exploring new code or developing your own work as you go.
+There are a slew of built-in helpers that we will go over in this lesson.
+
+### Autocomplete
+When working in the shell, you often might find yourself using a new module that you are unfamiliar with.
+To understand some of what is available to you, the autocomplete functionality is wonderful.
+Simply type a module name and then `.` and press `Tab`:
+
+```elixir
+iex> Map. # press Tab
+delete/2             drop/2               equal?/2
+fetch!/2             fetch/2              from_struct/1
+get/2                get/3                get_and_update!/3
+get_and_update/3     get_lazy/3           has_key?/2
+keys/1               merge/2              merge/3
+new/0                new/1                new/2
+pop/2                pop/3                pop_lazy/3
+put/3                put_new/3            put_new_lazy/3
+split/2              take/2               to_list/1
+update!/3            update/4             values/1
+```
+
+And now we know the functions we have and their arity!
+
+### `.iex.exs`
+Every time IEx starts, it will look for a `.iex.exs` file.
+If this is not present in the current directory, it will look at `~/.iex.exs` as a fallback.
+Anything inside it will be loaded in the shell that is started up.
+Lets say for some reason we really wanted to have a function called `fuzzy_bunnies/0` to be available all the time for some reason in a new project in a given namespace and have it print a message to us when we start our shell.
+
+We could simply do this:
+
+```shell
+$ mix new my_project
+$ cd my_project
+```
+
+And now we set it up:
+
+```elixir
+defmodule Fuzzy do
+  def fuzzy_bunnies do
+    IO.puts "FUZZY BUNNIES"
+  end
+end
+Fuzzy.fuzzy_bunnies
+```
+
+And when we run IEx:
+
+```shell
+$ iex
+Erlang/OTP 19 [erts-8.1] [source] [64-bit] [smp:4:4] [async-threads:10] [hipe] [kernel-poll:false] [dtrace]
+
+Interactive Elixir (1.3.3) - press Ctrl+C to exit (type h() ENTER for help)
+fuzzy bunnies
```

Comment:
Actually this is output from the console because we called `IO.puts Foo.hello_bunnies` [here](https://github.com/doomspork/elixir-school/pull/824/files#diff-8d8e482fd2b5c5092795333132ca88a3R58)